### DISCUSSION TITLE
Self complete formwidget

### DISCUSSION THREAD

**Discussion by damsfx** [wintercms/winter#150] [Show and tell]
A "text" formwidget which, like a "dropdown" field, offers options based on existing DB values.

![Formwidget Selfcomplete in action](https://github.com/hounddd/wn-selfcomplete-plugin/blob/master/docs/assets/Selfcomplete.gif?raw=true)

https://github.com/Hounddd/wn-selfcomplete-plugin

This field will act as a text entry letting you enter any value consistent with the field, but at the same time will provide you with the previous unique values for that same field.

**Comment by LukeTowers** [comment]
@damsfx doesn't the taglist already provide some level of this?

**Reply by damsfx** [reply]
The taglist field only provides the options that we define in the definition.
And it must be stored in a json field.

This field reads the data already present in the DB.

To illustrate, I made a festival management plugin, the users have to enter the location information of a show.
For example, the same city was written (in french) "Lons le saunier" or "lons-le-saunier".
If the entry is used in frontend to create filters, I have 2 times the same city with two different spellings.
The worst happens when they specify the room of the show: "room 1", "room-1", "room 1 - second floor" or even "room 1 / second floor".

By offering them the previous entries, it shows a better homogeneity in the data.

**Reply by LukeTowers** [reply]
@damsfx try this:

```
city:
    label: City SelfComplete
    mode: relation
    customTags: true
```

**Reply by damsfx** [reply]
@LukeTowers if I use the `taglist` field, need it to have a maxItems property set to 1 (no multi select) AND the options list to be dynamicely populated from exstiging data.

It's a bit of a special case
